scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/GraphQLQueries.java

```diff
@@ -147,111 +148,181 @@ private JobInput getJobInput(List<String> statusNames, String user, List<Constra
+                            user = filter;
+                        break;
+                    }
+
+                    case NAME: {
+
+                        if (constraint.getAction() == Action.EQUALS) {
+                            if (name == null)
+                                name = value;
+                            else if (value != name)
+                                return input.jobName(RETURN_NOTHING_FILTER).build();
+                        } else if ((filter = getFilter(constraint, value)) != null)
+                            name = filter;
+                        break;
+                    }
+
+                    case PROJECT_NAME: {
+
+                        if (constraint.getAction() == Action.EQUALS) {
+                            if (projectName == null)
+                                projectName = value;
+                            else if (value != projectName)
+                                return input.jobName(RETURN_NOTHING_FILTER).build();
+                        } else if ((filter = getFilter(constraint, value)) != null)
+                            projectName = filter;
+                        break;
+                    }
+
+                    case SUBMITTED_TIME: {
+
+                        try {
+
+                            dateInMs = ISODateTimeFormat.dateTimeParser().parseDateTime(value).toDate().getTime();
+
+                            switch (constraint.getAction()) {
+                                case GREATER_THAN_OR_EQUAL_TO:
+                                    if (afterSubmittedTime == -1 || dateInMs > afterSubmittedTime)
+                                        afterSubmittedTime = dateInMs;
+                                    break;
+                                case LESS_THAN_OR_EQUAL_TO:
+                                    if (beforeSubmittedTime == -1 || dateInMs < beforeSubmittedTime)
+                                        beforeSubmittedTime = dateInMs;
+                                    break;
+                                default:
+                                    break;
+                            }
+                        } catch (Exception e) {
+                            LOGGER.log(Level.SEVERE, "Error when parsing date filter \"" + value + "\"", e);
+                        }
+                        break;
                     }
-                    break;
                 }
             }
+
         }
 
-        if (user != null)
-            input.owner(user);
+        // Update the job input object and return it
         if (id != null)
             input.id(id);
+        if (afterId != null)
+            input.afterId(afterId);
+        if (beforeId != null)
+            input.beforeId(beforeId);
+        if (status != null)
+            input.status(status);
         if (priority != null)
             input.priority(priority);
+
+        if (userFilter != null)
```

Comment:
I didn't get this condition. "userFilter" is always null. You didn't assign any value to it.

Reply:
Fixed. Thx again.